### PR TITLE
refactor(example): remove @ngrx/db

### LINKED ISSUE #1481: Example: Remove @ngrx/db
Its not being maintained and there are better solutions for persisting state across page loads.

## If accepted, I would be willing to submit a PR for this feature

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

### PR DESCRIPTION
**Pull request by King-i**
add localStorage replacement

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #1481 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/example-app-cypress/integration/round-trip.spec.ts [hunk 1]
```diff
@@ -1,14 +1,34 @@
+let LOCAL_STORAGE_MEMORY = {};
+
+Cypress.Commands.add('saveLocalStorage', () => {
```
Comment (timdeschryver) [inline projects/example-app-cypress/integration/round-trip.spec.ts]: Add these to the commands at https://github.com/ngrx/platform/blob/master/projects/example-app-cypress/support/index.js

File: projects/example-app/src/app/core/services/book-storage.service.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  Inject,
+} from '@angular/core';
+import { Observable, of, throwError } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+import { Book } from '@example-app/books/models/book';
+
+export interface IBookStorageService {
+  supported(): Observable<boolean>;
+  deleteCollection(): Observable<boolean>;
+  addToCollection<T = Book>(records: T[]): Observable<T[]>;
```
Comment (timdeschryver) [inline projects/example-app/src/app/core/services/book-storage.service.ts]: This doesn't need to be a generic

File: projects/example-app-cypress/integration/round-trip.spec.ts [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
     (cy as any).eyesOpen({
       appName: 'books_app',
       testName: 'round-trip',
       browser: { width: 800, height: 600 },
     });
-    window.indexedDB.deleteDatabase('books_app');
+    window.localStorage.removeItem('books_app');
     cy.visit('/');
   });
+  beforeEach(() => {
```
Comment (timdeschryver) [inline projects/example-app-cypress/integration/round-trip.spec.ts]: Is the `beforeEach` and `afterEach` hook needed?

File: projects/example-app/src/app/books/effects/collection.effects.ts [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
 import {
-  SelectedBookPageActions,
-  CollectionPageActions,
   CollectionApiActions,
+  CollectionPageActions,
+  SelectedBookPageActions,
 } from '@example-app/books/actions';
+import { Book } from '@example-app/books/models/book';
+import { BookStorageService } from '@example-app/core/services/book-storage.service';
+import { Actions, Effect, ofType } from '@ngrx/effects';
```
Comment (timdeschryver) [inline projects/example-app/src/app/books/effects/collection.effects.ts]: Keep the import order as before, import `rxjs` and `@ngrx` packages before exame-app imports.

File: projects/example-app/src/app/core/services/book-storage.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+      providers: [
+        {
+          provide: LOCAL_STORAGE_TOKEN,
+          useValue: localStorageFake,
+        },
+      ],
+    });
+    fixture = TestBed.get(BookStorageService);
+  });
+  describe('supported', () => {
```
Comment (timdeschryver) [inline projects/example-app/src/app/core/services/book-storage.service.spec.ts]: Nit, add a line break between `describe`s / `it`s

File: projects/example-app/src/app/core/services/book-storage.service.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+export class BookStorageService implements IBookStorageService {
+  private collectionKey = 'books-app';
+
+  supported(): Observable<boolean> {
+    return this.storage !== undefined
+      ? of(true)
+      : throwError('Local Storage Not Supported');
+  }
+
+  getCollection<T = Book>(): Observable<T[]> {
```
Comment (timdeschryver) [inline projects/example-app/src/app/core/services/book-storage.service.ts]: This doesn't have to be a generic

File: projects/example-app-cypress/integration/round-trip.spec.ts (hunk 3 above)
Comment (King-i) [reply projects/example-app-cypress/integration/round-trip.spec.ts]: Required to persist the `localStorage` between the tests, if not we would have to duplicate test code to add books to the collection as `cypress` clears the `localStorage` after each test case

File: projects/example-app/src/app/books/effects/collection.effects.ts (hunk 4 above)
Comment (King-i) [reply projects/example-app/src/app/books/effects/collection.effects.ts]: `VsCode` default format on save by the looks of it :( Ill revert the order change

File: projects/example-app/src/app/core/services/book-storage.service.spec.ts (hunk 5 above)
Comment (King-i) [reply projects/example-app/src/app/core/services/book-storage.service.spec.ts]: Formatting is default `vscode` i believe, it also matches all other `spec` formatting in the project?

Comment (timdeschryver) [reply projects/example-app/src/app/core/services/book-storage.service.spec.ts]: I'm not sure if VSCode does this 🤔

Comment (King-i) [reply projects/example-app/src/app/core/services/book-storage.service.spec.ts]: Apologies you mentioned between `describe`s / `it`s, which it must do as I use a vanilla install of `vscode` with default settings, unless its the `prettier` formatter? if it was between the `beforeEach` and `describe` you intended then i have already actioned. I also believe the project runs a pre-commit formatting task.

File: projects/example-app/src/app/core/services/index.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export {
+  BookStorageService,
+  LOCAL_STORAGE_TOKEN,
+} from '@example-app/core/services/book-storage.service';
```
Comment (brandonroberts) [inline projects/example-app/src/app/core/services/index.ts]:
```suggestion
} from './book-storage.service';
```

**Comment by brandonroberts** [comment]
@King-i this needs a rebase on master to fix the conflicts from adding the `createEffect` function

**Comment by brandonroberts** [comment]
LGTM. Needs approval from @timdeschryver for his review comments.

**Pull request merged**